packages/react/src/utils/positioner/Popper.tsx

```diff
@@ -58,18 +78,59 @@ const Popper: React.FunctionComponent<PopperProps> = props => {
 
   const popperRef = React.useRef<PopperJS>()
   const contentRef = React.useRef<HTMLElement>(null)
+
   const latestPlacement = React.useRef<PopperJS.Placement>(proposedPlacement)
   const [computedPlacement, setComputedPlacement] = React.useState<PopperJS.Placement>(
     proposedPlacement,
   )
 
-  const computedModifiers: PopperJS.Modifiers = React.useMemo(
+  const hasScrollableElement = React.useMemo(() => {
+    const scrollParentElement = getScrollParent(contentRef.current)
+
+    return scrollParentElement !== scrollParentElement.ownerDocument.body
+  }, [contentRef])
+
+  const computedModifiers: PopperJS.Modifiers = useDeepMemo(
     () =>
-      offset && {
-        offset: { offset: rtl ? applyRtlToOffset(offset, position) : offset },
-        keepTogether: { enabled: false },
-      },
-    [rtl, offset, position],
+      _.merge(
+        /**
+         * This prevents blurrines in chrome, when the coordinates are odd numbers alternative
+         * would be to use `fn` and manipulate the computed style or ask popper to fix it but
+         * since there is presumably only handful of poppers displayed on the page, the
+         * performance impact should be minimal.
+         */
+        { computeStyle: { gpuAcceleration: false } },
+
+        { flip: { padding: 0, flipVariationsByContent: true } },
+        { preventOverflow: { padding: 0 } },
+
+        offset && {
+          offset: { offset: rtl ? applyRtlToOffset(offset, position) : offset },
+          keepTogether: { enabled: false },
+        },
+
+        /**
+         * When the popper box is placed in the context of a scrollable element, we need to set
+         * preventOverflow.escapeWithReference to true and flip.boundariesElement to 'scrollParent'
+         * (default is 'viewport') so that the popper box will stick with the targetRef when we
+         * scroll targetRef out of the viewport.
+         */
+        hasScrollableElement && {
+          preventOverflow: { escapeWithReference: true },
+          flip: { boundariesElement: 'scrollParent' },
+        },
+
+        userModifiers,
+
+        /**
+         * unstable_pinned disables the flip modifier by setting flip.enabled to false; this
+         * disables automatic repositioning of the popper box; it will always be placed according to
+         * the values of `align` and `position` props, regardless of the size of the component, the
+         * reference element or the viewport.
+         */
+        unstable_pinned && { flip: { enabled: false } },
+      ),
+    [hasScrollableElement, position, offset, rtl, unstable_pinned, userModifiers],
```

Comment:
We have `userModifiers` as dependency there as they are an object, to avoid recreation of `Popper` on each render I decided to use there deep memo strategy.